C/plugins/storage/postgres/connection.cpp

```diff
@@ -3312,3 +3314,712 @@ const char *p;
+                                        return -1;
+				}
+
+				if (!document.HasMember("version"))
+                        	{
+					Logger::getLogger()->error("version absent from payload");
+                                	return -1;
+                        	}
+				else
+				{
+					version = document["version"].GetInt();
+
+					if (!schema.empty())
+					{
+						std::string results;
+						if (findSchemaFromDB(service, schema, results))
+						{
+							if (!parseDatabaseStorageSchema(version, results, columnMapFromDB, indexMapFromDB, tbKeyMapFromDB, schemaCreationReq))
+							{
+								Logger::getLogger()->error("error in parsing Database Storage schema");
+								return -1;
+							}
+						}
+							
+						std::string queryToCreateSchema = "create schema if not exists " + schema + ";" ;
+	                                        rowsAffectedLastCommand = purgeOperation(queryToCreateSchema.c_str(), logSection, "Create Schema if not exists ", false);
+						if (rowsAffectedLastCommand == -1)
+						{
+							Logger::getLogger()->error("%s:%d Error in creating schema",__FUNCTION__,__LINE__);
+							return -1;
+						}
+					}
+
+				}
+				if (!document.HasMember("tables"))
+                        	{
+					Logger::getLogger()->error("tables section absent from payload");
+                                	return -1;
+                        	}
+				else
+				{
+					Value& tables = document["tables"];
+					if (!tables.IsArray())
+                                	{
+                                        	Logger::getLogger()->error("The property return must be an array");
+                                        	return -1;
+                                	}
+					else
+                          		{
+						std::unordered_set<std::string> unSetTablesInSchemaRequest;
+						std::string sqlDropTables;
+
+						// Iterate over all the table lists in the Schema Creation/Alter request
+                                		for (rapidjson::SizeType i = 0; i < tables.Size(); i++)
+						{
+							std::string name = tables[i]["name"].GetString();
+							unSetTablesInSchemaRequest.insert(name);
+							Value& columns = tables[i]["columns"];
+							std::vector<std::string> indexesMatrixFromReq;
+							std::unordered_set<columnRec, columnRecHasher, columnRecComparator> colsPerTableInReq;
+							bool alterTable = false;
+							std::string sql, sqlIdx;
+
+							// if this is schema creation request  or 
+							// this table does not exist in db, then create it 
+							// else alter the table
+							if (schemaCreationReq || (columnMapFromDB.find(name) == columnMapFromDB.end()))
+							{
+								sql = "create table " + schema + "." + name + " (" ;
+							}
+							else
+							{
+								sql = "alter table " + schema + "." + name + " " ;
+								alterTable = true;
+							}
+					
+							if (!columns.IsArray())
+							{
+								Logger::getLogger()->error("The property columns must be an array");
```

Comment:
Context and content of the error message please.

Reply:
Fixed , added a debug log which will mention about which  table getting parsed